server/py/services/alerts/crud/alerts.py

```diff
@@ -175,76 +175,34 @@ def process_event(
-                    # in case the EventEntityKind is JOB then we should consider the runs monitoring interval here
-                    # because the monitoring runs might miss events occurring just before the interval.
-                    offset = 0
-                    if (
-                        alert.entities.kind
-                        == mlrun.common.schemas.alert.EventEntityKind.JOB
-                    ):
-                        offset = int(mlconfig.monitoring.runs.interval)
-                    self._normalize_events(
-                        state_obj,
-                        framework.utils.helpers.string_to_timedelta(
-                            alert.criteria.period, offset, raise_on_error=False
-                        ),
-                    )
+        if not self._event_entity_matches(alert.entities, event_data.entity):
+            return
 
-                if len(state_obj["events"]) >= alert.criteria.count:
-                    send_notification = True
-            else:
-                send_notification = True
-
-            active = False
-            update_state = True
-            if send_notification:
-                state["count"] += 1
-                logger.debug("Sending notifications for alert", name=alert.name)
-                AlertNotificationPusher().push(alert, event_data)
-
-                if alert.reset_policy == "auto":
-                    self.reset_alert(session, alert.project, alert.name)
-                    services.api.crud.AlertActivation().store_alert_activation(
-                        session, alert, event_data
-                    )
-                    update_state = False
-                else:
-                    active = True
-                    state["active"] = True
-                    self._get_alert_state_cached().cache_replace(
-                        state, session, alert.id
-                    )
+        state_obj = self._states.get(alert.id, {"events": []})
 
-                # we store the state along with the events that triggered the alert
-                framework.utils.singletons.db.get_db().store_alert_state(
-                    session,
-                    alert.project,
-                    alert.name,
-                    count=state["count"],
-                    last_updated=event_data.timestamp,
-                    obj=state_obj,
-                    active=active,
-                )
+        # Append event timestamp for active state or criteria
+        if state["active"] or alert.criteria is not None:
+            state_obj["events"].append(event_data.timestamp)
 
-            if update_state:
-                # we don't want to update the state if reset_alert() was called, as we will override the reset
-                self._states[alert.id] = state_obj
+        # Exit early if state is active (no further processing needed)
+        if state["active"]:
+            self._states[alert.id] = state_obj
+            return
```

Comment:
So to clarify - the number of events on the activation is not incremented every event, but will stay the same until the alert is reset? even if more events are still coming in?
Won't it make more sense to update it on each event?

Reply:
@TomerShor it will create more load on database. Not that it's not feasible, but for now we decided to do like this (please check @theSaarco comment in slack)

Reply:
Correct. We certainly want to avoid going to the DB on every event (after all, that's why we keep cache).